addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -1903,16 +1903,64 @@ var SnippetsMenu = Widget.extend({
 
             $invisibleDOMPanelEl.toggleClass('d-none', !$invisibleSnippets.length);
 
-            const proms = _.map($invisibleSnippets, async el => {
-                const editor = await this._createSnippetEditor($(el));
-                const $invisEntry = $('<div/>', {
-                    class: 'o_we_invisible_entry d-flex align-items-center justify-content-between',
-                    text: editor.getName(),
-                }).append($('<i/>', {class: `fa ${editor.isTargetVisible() ? 'fa-eye' : 'fa-eye-slash'} ml-2`}));
-                $invisibleDOMPanelEl.append($invisEntry);
-                this.invisibleDOMMap.set($invisEntry[0], el);
+            // descendantPerSnippet: a map with its keys set to invisible
+            // snippets that have invisible descendants. The value
+            // corresponding to an invisible snippet element is a list filled
+            // with all its descendant invisible snippets except those that
+            // have a closer invisible snippet ancestor.
+            const descendantPerSnippet = new Map();
+            // Filter the "$invisibleSnippets" to only keep the root snippets
+            // and create the map ("descendantPerSnippet") of the snippets and
+            // their descendant snippets.
+            const rootInvisibleSnippetEls = [...$invisibleSnippets].filter(invisibleSnippetEl => {
+                const ancestorInvisibleElement = invisibleSnippetEl
+                                                 .parentElement.closest(".o_snippet_invisible");
+                if (ancestorInvisibleElement) {
+                    const descendantSnippets = descendantPerSnippet
+                                               .get(ancestorInvisibleElement) || [];
+                    descendantPerSnippet
+                    .set(ancestorInvisibleElement, [...descendantSnippets, invisibleSnippetEl]);
+                    return false;
+                }
+                return true;
```

Comment:
> I adapted the code to avoid too long line. However, isn't this line for example a bit less readable like this?

Make it that way will help :+1: 

```js
if (!ancestorInvisibleElement) {
    return true;
}
const descendantSnippets = descendantPerSnippet.get(ancestorInvisibleElement) || [];
descendantPerSnippet.set(ancestorInvisibleElement,
    [...descendantSnippets, invisibleSnippetEl]);
return false;
```